src/azure-cli/azure/cli/command_modules/botservice/channel_operations.py

```diff
@@ -19,6 +23,25 @@ def create_channel(client, channel, channel_name, resource_group_name, resource_
     )
 
 
+def update_channel(client, channel, channel_name, resource_group_name, resource_name):
+    bot_channel_data = client.get(
+        resource_group_name,
+        resource_name,
+        channel_name)
+    sites = bot_channel_data.properties.properties.sites
+
+    for site in sites:
+        if site.site_name == channel.properties.sites[0].site_name:
+            channel.properties.sites[0].site_id = site.site_id
+
+    return client.update(
+        resource_group_name=resource_group_name,
+        resource_name=resource_name,
+        channel_name='DirectLineChannel',
```

Comment:
Use channel_name=channel_name instead of channel_name='DirectLineChannel' to make update_channel to be a more generic method?

Reply:
We don't currently support updating other channels, but I've applied this feedback.